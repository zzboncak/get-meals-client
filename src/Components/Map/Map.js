import React from 'react'
import MapGoogle from './WrappedMap'
import './Map.css';


export default function Maps(props) {

  return (
    <div className="mapContainer">
      {/** The css must be explicit for these cases! */}
      <MapGoogle
        googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=
        ${process.env.REACT_APP_GOOGLE_KEY}
        `}
        loadingElement={<div style={{ height: "100%" }} />}
        containerElement={<div style={{ height: "400px" }} />}
        mapElement={<div style={{ height: "100%" }} />}
        locations={props.locations}

      />
    </div>
  )
}