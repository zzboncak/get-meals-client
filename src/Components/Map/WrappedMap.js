import React from "react";
import { withScriptjs, withGoogleMap, GoogleMap, Marker } from "react-google-maps"


const MapGoogle = withScriptjs(
    withGoogleMap(props => {
        return (
            <>
                <GoogleMap
                    defaultZoom={4}
                    defaultCenter={{ lat: 37.0902, lng: -95.7129 }} >


                </GoogleMap>
                

                {props.locations.map(coordinates => (
                    <Marker 
                        key={coordinates.id}
                        position={{
                            lat: coordinates.location_latitude,
                            lng: coordinates.location_longitude
                        }}
                    />
                ))}
            </>
        )
    })
)

export default MapGoogle