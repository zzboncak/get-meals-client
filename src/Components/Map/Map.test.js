import React from 'react';
import ReactDOM from 'react-dom';
import Map from './Map';

describe(`Map component`, () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Map />, div);
      ReactDOM.unmountComponentAtNode(div);
  });
})