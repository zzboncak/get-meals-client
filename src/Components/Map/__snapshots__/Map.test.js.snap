// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component renders the component as expected 1`] = `
<div
  className="mapContainer"
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
</div>
`;
